controllers/mirrorpeer_controller.go

```diff
@@ -106,6 +108,19 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 	logger.V(2).Info("All validations for MirrorPeer passed", "MirrorPeer", req.NamespacedName)
 
+	if mirrorPeer.Labels == nil {
```

Comment:
Since we are reading an object from cache and trying to update it, we need to DeepCopy() and work on the copy of object. Cache will then get updates from apiserver on next refresh.

```suggestion
        mirrorPeerCopy := mirrorPeer.DeepCopy()
	if mirrorPeerCopy.Labels == nil {
```